build_tools/third_party/cuda/CMakeLists.txt

```diff
@@ -6,16 +6,48 @@
 
 set(CUDA_HEADERS_API_ROOT "${IREE_ROOT_DIR}/third_party/cuda/include")
 
-external_cc_library(
-  PACKAGE
-    cuda_headers
-  NAME
-    cuda_headers
-  ROOT
-    ${CUDA_HEADERS_API_ROOT}
-  HDRS
-    "cuda.h"
-  INCLUDES
-    ${CUDA_HEADERS_API_ROOT}
-)
+if (TARGET CUDA::cupti)
```

Comment:
and runtime tracing enabled (to my other comment about downstream projects providing this target but not enabling IREE's tracing)

For some more background, see projects like https://dawn.googlesource.com/tint/+/refs/heads/main/third_party/CMakeLists.txt where they let downstream projects provide targets (from their own submodules or elsewhere) and if they aren't provided then they set their own.